compiler/rustc_resolve/src/ident.rs

```diff
@@ -974,7 +974,8 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
         // shadowing is enabled, see `macro_expanded_macro_export_errors`).
         let unexpanded_macros = !module.unexpanded_invocations.borrow().is_empty();
         if let Some(binding) = binding {
-            if !unexpanded_macros || ns == MacroNS || restricted_shadowing {
+            let determined = !unexpanded_macros && binding.determined();
+            if determined || ns == MacroNS || restricted_shadowing {
```

Comment:
```suggestion
            if binding.determined() || ns == MacroNS || restricted_shadowing {
```
`module` above is the `binding`'s module, so the `!unexpanded_macros` should be already included and this should be equivalent.